### PR TITLE
Add documentation about using list of objects in @ConfigProperties

### PR DESCRIPTION
**Pull request by geoand**
Related to: #14006

### CODE REVIEW

**Review by emretanriverdi — APPROVED** [review]

File: docs/src/main/asciidoc/config-reference.adoc [hunk 1]
```diff
@@ -310,6 +310,52 @@ public class SomeBean {
...
+    public List<Nested> outputs;
+
+    public static class Nested {
+        public String user;
+        public String password;
+    }
+}
+----
+
+Support for such use cases is only possible when the `quarkus-config-yaml` extension is present. A corresponding example configuration could be:
```
Comment (gsmet) [inline docs/src/main/asciidoc/config-reference.adoc:335]: I think this is slightly misleading. I suppose it's only working if you are using Yaml configuration? The way it's worded, it looks like just having the extension around is enough.

**Review by gsmet — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/config-reference.adoc (hunk 1 above)
Comment (geoand) [reply docs/src/main/asciidoc/config-reference.adoc]: How about now?

Comment (gsmet) [reply docs/src/main/asciidoc/config-reference.adoc]: Mmmmh I don't see any difference?

Was thinking about something like:

> Support for such use cases is only available when using YAML configuration with the `quarkus-config-yaml` extension.

Comment (geoand) [reply docs/src/main/asciidoc/config-reference.adoc]: I don't see the difference TBH, but I'll just copy what you mention and be done with ti

Comment (geoand) [reply docs/src/main/asciidoc/config-reference.adoc]: Updated

**Review by gastaldi — APPROVED** [review]
LGTM

**Review by gsmet — APPROVED** [review]

**Pull request merged**